mobile/android/WirelessDebug/wirelessdebugger/src/main/java/com/google/wireless/debugger/WebSocketMessenger.java

```diff
@@ -122,17 +127,17 @@ public void sendLogDump() {
             return;
         }
 
+        // Copy the array list to prevent possible race conditions when sending/enqueuing logs
+        ArrayList<String> logsToSendCopy = new ArrayList<>(mLogsToSend);
```

Comment:
I think it is thread safe already.  LogReader and WebSocketMessenger run on the same thread. The actual sending of the messages takes place on a separate thread, and the object being sent is a String made from the JSON object.